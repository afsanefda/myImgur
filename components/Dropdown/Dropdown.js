// @flow
import React from 'react';

type Props = {
  defaultValue:string,
  list: Array<string>,
  handleChangeFilter: Function
};

const Dropdown = ({
  defaultValue, list, handleChangeFilter,
}: Props) => {
  const handleChange = (event) => {
    handleChangeFilter(event.target.value);
  };

  return (
    <div className="DropdownContainer">
      <select value={defaultValue} onChange={handleChange}>
        {list?.map((el) => <option key={el.value} value={el.value}>{el.value}</option>)}
      </select>
    </div>
  );
};
export default Dropdown;
