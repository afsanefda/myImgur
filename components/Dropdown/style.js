import styled from 'styled-components';

export const DropdownContainer = styled.div`
  display: flex;
  margin: 5px;
  justify-content: space-between;
  align-items: center;
`;
export const DropdownSelect = styled.select`
  background: #F58535;
  border: none;
  color: #fff;
  border-radius: 6px;
  padding: 5px 10px;
  height: 26px;
  width: 100px;
  &:focus {
    outline: none;
    }
`;
