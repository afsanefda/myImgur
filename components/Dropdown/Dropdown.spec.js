import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

describe('<Dropdown/>', () => {
  let rootElement;
  let testRenderer;

  // eslint-disable-next-line prefer-const
  rootElement = () => (
    <Dropdown />
  );
  // eslint-disable-next-line prefer-const
  testRenderer = renderer.create(rootElement());

  it('Should render correctly', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
