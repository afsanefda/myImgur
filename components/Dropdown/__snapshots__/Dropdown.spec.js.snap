// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dropdown should render 1`] = `"div"`;
